lib/src/json_differ.dart

```diff
@@ -5,30 +5,50 @@ part of json_diff;
 
 /// A configurable class that can produce a diff of two JSON Strings.
 class JsonDiffer {
-  Map<String,Object> leftJson, rightJson;
-  final List<String> atomics = new List<String>();
-  final List<String> metadataToKeep = new List<String>();
-  final List<String> ignored = new List<String>();
-  
+  Map<String, Object> leftJson, rightJson;
+  final List<String> atomics = <String>[];
+  final List<String> metadataToKeep = <String>[];
+  final List<String> ignored = <String>[];
+  bool debug;
+
   /// Constructs a new JsonDiffer using [leftString] and [rightString], two
   /// JSON objects represented as Dart strings.
   ///
   /// If the two JSON objects that need to be diffed are only available as
   /// Dart Maps, you can use the
   /// [dart:convert](https://api.dartlang.org/apidocs/channels/stable/dartdoc-viewer/dart:convert)
   /// library to encode each Map into a JSON String.
-  JsonDiffer(leftString, rightString) {
-    Object _leftJson = new JsonDecoder().convert(leftString);
-    Object _rightJson = new JsonDecoder().convert(rightString);
-    
-    if (_leftJson is Map && _rightJson is Map) {
+  JsonDiffer(
+    String leftString,
+    String rightString, {
+    bool debug = false,
```

Comment:
This new parameter should be documented.

Reply:
I'll remove it altogether.